includes/helpers.php

```diff
@@ -155,3 +157,61 @@ function get_preferred_models(): array {
 	 */
 	return (array) apply_filters( 'ai_preferred_models', $preferred_models );
 }
+
+/**
+ * Checks if we have AI credentials set.
+ *
+ * @since 0.1.0
+ *
+ * @return bool True if we have AI credentials, false otherwise.
+ */
+function has_ai_credentials(): bool {
+	$credentials = get_option( 'wp_ai_client_provider_credentials', array() );
+
+	// If there are no credentials, return false.
+	if ( ! is_array( $credentials ) || empty( $credentials ) ) {
+		return false;
+	}
+
+	// If all of the AI keys are empty, return false; otherwise, return true.
+	return ! empty(
+		array_filter(
+			$credentials,
+			static function ( $api_key ): bool {
+				return is_string( $api_key ) && '' !== $api_key;
+			}
+		)
+	);
+}
+
+/**
+ * Checks if we have valid AI credentials.
+ *
+ * @since 0.1.0
+ *
+ * @return bool True if we have valid AI credentials, false otherwise.
+ */
+function has_valid_ai_credentials(): bool {
+	/**
+	 * Filters whether valid AI credentials are available.
+	 *
+	 * Allows overriding the credentials check, useful for testing.
+	 *
+	 * @since 0.1.0
+	 * @hook ai_pre_has_valid_credentials_check
+	 *
+	 * @param bool|null $has_valid_credentials Whether valid credentials are available. Return null to use default check.
+	 * @return bool|null True if valid credentials are available, false otherwise, or null to use default check.
+	 */
+	$valid = apply_filters( 'ai_pre_has_valid_credentials_check', null );
+	if ( null !== $valid ) {
+		return (bool) $valid;
+	}
+
+	// If we have no AI credentials, return false.
+	if ( ! has_ai_credentials() ) {
```

Comment:
Shouldn't this be called before the filter? If you don't have _any_ AI credentials, you certainly don't have _valid_ AI credentials.

It makes sense to me to allow filtering/overriding the validity check logic, but the prerequisite should always be that there are AI credentials at all.

Otherwise, this also throws a wrench in things when e.g. for the admin notice you're also checking `has_ai_credentials()`, which would lead to a misleading notice if that filter here was overriding the result to be `true`.